src/modules/pivotTable/renderValue.js

```diff
@@ -0,0 +1,49 @@
+const DEFAULT_PRECISION = 1,
+    SMALL_NUMBER_PRECISION = 2,
+    SKIP_ROUNDING_PRECISION = 10
+
+const separateDigitGroups = (stringValue, decimalSeparator = '.') => {
+    const integerLength =
+        stringValue.indexOf('.') !== -1
+            ? stringValue.indexOf('.') + 1
+            : stringValue.length
+    const remainder = stringValue.substring(integerLength)
+
+    const groups = []
+    for (let i = integerLength - 1; i > 0; i -= 3) {
+        groups.unshift(stringValue.substring(i - 3, i))
+    }
+
+    if (remainder.length) {
+        groups[groups.length - 1] += decimalSeparator + remainder
+    }
+
+    return groups
+}
+
+export const renderValue = (value, visualization) => {
+    const precision = visualization.skipRounding
+        ? SKIP_ROUNDING_PRECISION
+        : value < 1 && value > -1
+        ? SMALL_NUMBER_PRECISION
+        : DEFAULT_PRECISION
+
+    if (isNaN(value)) {
+        return value
+    }
+
+    const stringValue = parseFloat(value.toFixed(precision)).toString() // TODO: This rounds and maybe it shouldn't!
```

Comment:
It does, but then we round again to a certain number of decimal points here - if the precision of the number sent by the backend doesn't exactly match the hard-coded precision here (1, 2, or 10 decimal points) then the number will be rounded again (even with SKIP_ROUNDING).  This was the existing behavior, just made a not that we might want to truncate instead of rounding here.

Reply:
We actually don't want the existing behaviour anymore. Backend should handle precision as well now. We simply shouldn't touch the values. I suggest a backport to 2.33 to align with the new behaviour.

Reply:
Not sure it's needed, the rounding shouldn't happen if the backend and frontend have the same precisions (as they do in 2.33).  We still need to process the numbers though to parse them for totals calculation as well as to trim trailing 0s.  Opened #298

Reply:
Yeah, good point.